In [4]:
import pandas as pd

The following dataset has been downloaded from kaggle.com. This dataset contains information collected in the area of Boston.

In [11]:
Boston = pd.read_csv('boston_data.csv')

In [10]:
Boston.head()

crim    zn  indus  chas    nox     rm   age     dis  rad  tax  ptratio  \
0  0.15876   0.0  10.81     0  0.413  5.961  17.5  5.2873    4  305     19.2   
1  0.10328  25.0   5.13     0  0.453  5.927  47.2  6.9320    8  284     19.7   
2  0.34940   0.0   9.90     0  0.544  5.972  76.7  3.1025    4  304     18.4   
3  2.73397   0.0  19.58     0  0.871  5.597  94.9  1.5257    5  403     14.7   
4  0.04337  21.0   5.64     0  0.439  6.115  63.0  6.8147    4  243     16.8   

    black  lstat  medv  
0  376.94   9.88  21.7  
1  396.90   9.22  19.6  
2  396.24   9.97  20.3  
3  351.85  21.45  15.4  
4  393.97   9.43  20.5

In [13]:
Boston.shape

(404, 14)

Data frame has 404 rows with 14 columns

### Creating depedent and independent variable

In [8]:
y = Boston[['medv']]

medv column has been selected as dependent variable.

In [9]:
x = Boston[['crim']]

crim column has been selected as independent variable.

### Importing linear regression from sklearn

In [8]:
from sklearn.linear_model import LinearRegression

### Split the data in train and test data

In [11]:
from sklearn.model_selection import train_test_split

In [14]:
x_train, x_test,y_train,y_test= train_test_split(x,y,test_size=0.3)

The dataset has been split into 30% and 70% in test and train data respectively

In [16]:
lr = LinearRegression()

#### Fit the model on training data

In [18]:
lr.fit(x_train,y_train)

LinearRegression()

In [19]:
y_pred = lr.predict(x_test)

In [21]:
y_test.head()

medv
183  21.7
247  20.7
262  21.2
382  50.0
6    26.2

In [23]:
y_pred[0:5]

array([[23.60079408],
       [23.62015699],
       [23.61360656],
       [21.38814162],
       [23.56750977]])

In [25]:
from sklearn.metrics import mean_squared_error

In [27]:
mean_squared_error(y_test,y_pred)

69.05014382562953

In [28]:
x = Boston[['indus']]

In [31]:
x_train,x_test,y_train,y_test = train_test_split(x,y,test_size=0.3)

In [32]:
lr2 = LinearRegression()

In [33]:
lr2.fit(x_train,y_train)

LinearRegression()

In [34]:
y_pred2 = lr.predict(x_test)

In [35]:
from sklearn.metrics import mean_squared_error

In [38]:
mean_squared_error(y_test, y_pred2)

78.70734636798969